regression_and_census.R

```diff
@@ -9,9 +9,169 @@
 # old code-  I M P O R T I N G   C E N S U S   D A T A ---------------------------------------
 
 library(latticeExtra)
-install.packages("gee")
+library(gee)
+library(geepack)
+library(MASS)
+library(broom)
 # tidyverse
 library(dplyr)
+library(readr)
+library(tidyr)
+library(magrittr)
+library(stringr)
+library(purrr)
+# plotting libraries
+library(ggplot2)
+library(ggmap)
+library(gridExtra)
+
+
+
+# Load data ---------------------------------------------------------------
+
+clean_weekly_data_dir <- "." #"/Users/funmiarogbokun/Desktop/CZ Biohub Contractor/Wildfires & Covid Project/Manuscript/Clean_final_datasets/Weekly_data"
+CA_PM25_wk_PM <- read_csv(file.path(clean_weekly_data_dir, "CA_weekly_lagged_averages.csv")) %>%
+  mutate(county=factor(county))
+
+census_data_dir <- "." #"/Users/funmiarogbokun/Desktop/CZ Biohub Contractor/Wildfires & Covid Project/Census"
+census_counties_2020 <- read_csv(file.path(census_data_dir, "census_CAcountydata_2020.csv"))
+
+
+# Clean and transform data for California ---------------------------------
+
+#merging the two datasets
+pol_covid_census_2020 <- CA_PM25_wk_PM %>%
+  left_join(census_counties_2020, by=c("county"="BASENAME")) %>%
+  # creating pop density variable from census data
+  mutate(pop_density_new=POP100 /AREALAND)
+
+
+# explanatory and response variables
+explanatory_var <- "PM25_wk_avg"
+# This line checks the column names in CA_PM25_wk_PM to identify which columns
+# are response variables: those with 'lag_wkg' in the colum name)
+response_vars <- grep("lag_wkavg", names(CA_PM25_wk_PM), value=TRUE)
+covariates <- c("month", "pop_density_new", "P0020002", "P0020003", "P0020005",
+                "P0020006", "P0020007", "P0020008", "P0020009", "P0020010")
+
+
+# Create datasets to fit models to ----------------------------------------
+
+#UPDATE - data with cutoff at 30 PM
+PM_CA_all <- pol_covid_census_2020 %>% 
+  # Month should be a categorical variable, hence changing its type numeric --> factor.
+  mutate(month=factor(month)) %>%
+  filter(., complete.cases(.)) %>%
+  slice(-unique(which(select(., c(explanatory_var, response_vars, covariates)) < 0, arr.ind=TRUE)[, 1]))
+PM_CA_all[, response_vars] %<>% sapply(., as.integer)
+cols_to_standardize <- c("pop_density_new", grep("P00", covariates, value=TRUE))
+multipliers <- select(PM_CA_all, cols_to_standardize) %>% apply(2, max)
+PM_CA_all[, cols_to_standardize] %<>%
```

Comment:
I also normalized the population densities and other census variables by dividing the values by the value in that column. This is to standardize the values so they all fall in the 0 to 1 range. This helps models converge more easily as the algorithm doesn't have to search a wide parameter range.

You can get the original values back by multiplying each column by the value in `multipliers`.